README.md

```diff
@@ -0,0 +1,146 @@
+# Capsule sandbox
+
+The Capsule sandbox is a containerized development environment for building Capsule applications in Rust.
+
+## Table of Contents
+
+* [Introduction](#introduction)
+* [Getting Started](#getting-started)
+  * [Vagrant and VirtualBox](#vagrant-and-virtualbox)
+  * [Linux Distributions](#linux-distributions)
+  * [Without Docker](#without-docker)
+  * [Kernel NIC Interface](#kernel-nic-interface)
+* [Packaging for Release](#packaging-for-release)
+* [Contributing](#contributing)
+* [Code of Conduct](#code-of-conduct)
+* [License](#license)
+
+## Introduction
+
+`Capsule` is built on Intel's [Data Plane Development Kit](https://www.dpdk.org/) release 18.11. While it is written in Rust, it needs to be able to call `DPDK`'s C functions installed separately on the host as shared dynamic libaries. To make developing and running Capsule applications easy, we created the sandbox container with all the necessary dependencies. You can pull down the container and start writing network functions in stable `Rust` right away.
+
+## Getting Started
+
+To run the sandbox, the docker host must run a Linux distribution. `DPDK` requires either Linux or FreeBSD. We may add FreeBSD support in the future.
+
+### Vagrant and VirtualBox
+
+The quickest way to start the sandbox is to use the Vagrant virtual machine included in this repository. If you are developing on either MacOS or Windows, this is the only way to write a `Capsule` application.
+
+After cloning this repository, you can start the vagrant VM using commands,
+
+```
+host$ vagrant up docker
+host$ vagrant ssh
+```
+
+Now you are inside a `Debian` VM. The system is already preconfigured for DPDK. To run the sandbox, use the command,
+
+```
+vagrant$ docker run -it --rm \
+    --privileged \
+    --network=host \
+    --name sandbox \
+    --cap-add=SYS_PTRACE \
+    --security-opt seccomp=unconfined \
+    -v /lib/modules:/lib/modules \
+    -v /dev/hugepages:/dev/hugepages \
+    getcapsule/sandbox:18.11.6-1.42 /bin/bash
+```
+
+The sandbox must run in privileged mode with host networking, so it can access the two network interfaces on the Vagrant host dedicated to DPDK applications.
+
+```
+vagrant$ lspci
+00:08.0 Ethernet controller: Red Hat, Inc Virtio network device
+00:09.0 Ethernet controller: Red Hat, Inc Virtio network device
+```
+
+To use `Capsule`, add it as a dependency in your `Cargo.toml`,
+
+```
+[dependencies]
+capsule = "0.1"
+```
+
+You should also mount the working directory of your project as a volume for the sandbox. Then you can use `cargo` commands inside the container as normal.
```

Comment:
link to submod?

Reply:
as I was writing this, I realized the most common direction should actually target people writing projects using capsule as a dependency rather than contributing to capsule. that's why I didn't make assumptions here about they need to mount capsule, and just "project" in general.